frontend/src/views/SubmissionView.vue

```diff
@@ -88,17 +167,96 @@ onMounted(async () => {
       </span>
     </TabPanel>
     <TabPanel header="Files">
-      <div class="grid nested-grid">
-        <div class="col-2">
-          <FileUpload :activity-id="props.activityId" />
+      <div class="mb-3 border-dashed fileupload border-round-md">
+        <FileUpload
+          :activity-id="props.activityId"
+          class=""
+        />
+      </div>
+      <div class="flex flex-row justify-content-between pb-3">
+        <div class="flex align-items-center">
+          <IconField icon-position="left">
+            <InputIcon class="pi pi-search" />
+            <InputText
+              v-model="searchTag"
+              placeholder="Search"
+            />
+          </IconField>
         </div>
-        <div class="col-10">
+        <div class="align-items-end">
+          <Button
+            aria-label="List"
+            class="view-switch-button"
+            @click="gridView = false"
+          >
+            <font-awesome-icon
+              icon="fa-solid fa-list"
+              :class="gridView ? 'list-grid-deselected-icon' : 'list-grid-selected-icon'"
+            />
+          </Button>
+          <Button
+            aria-label="Grid"
+            class="view-switch-button"
+            @click="gridView = true"
+          >
+            <font-awesome-icon
+              icon="fa-solid fa-grip"
+              :class="gridView ? 'list-grid-selected-icon' : 'list-grid-deselected-icon'"
+            />
+          </Button>
+        </div>
+      </div>
+      <div
+        v-if="gridView"
+        class="grid nested-grid"
+      >
+        <DataTable
+          v-if="gridView"
+          class="remove-padding"
+          sort-field="createdAt"
```

Comment:
Per prior comment, remove magic strings. Check other similar props for the same.